sdk/core/azure-core/src/main/java/com/azure/core/util/implementation/FileContent.java

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util.implementation;
+
+import com.azure.core.util.FluxUtil;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.serializer.ObjectSerializer;
+import com.azure.core.util.serializer.TypeReference;
+import reactor.core.Exceptions;
+import reactor.core.publisher.Flux;
+import reactor.core.scheduler.Scheduler;
+import reactor.core.scheduler.Schedulers;
+
+import java.io.BufferedInputStream;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * A {@link BinaryDataContent} backed by a file.
+ */
+public class FileContent extends BinaryDataContent {
+    private final ClientLogger logger = new ClientLogger(FileContent.class);
+    private final Path file;
+    private final int chunkSize;
+    private final long length;
+    private final Scheduler scheduler;
+    private final AtomicReference<byte[]> bytes = new AtomicReference<>();
+
+    /**
+     * Creates a new instance of {@link FileContent}.
+     *
+     * @param file The {@link Path} content.
+     * @param chunkSize The requested size for each read of the path.
+     */
+    public FileContent(Path file, int chunkSize) {
+        Objects.requireNonNull(file, "'file' cannot be null.");
+
+        if (chunkSize <= 0) {
+            throw logger.logExceptionAsError(new IllegalArgumentException(
+                    "'chunkSize' cannot be less than or equal to 0."));
+        }
+        this.file = file;
+        this.chunkSize = chunkSize;
+        if (!file.toFile().exists()) {
+            throw logger.logExceptionAsError(new UncheckedIOException(
+                    new FileNotFoundException("File does not exist " + file)));
+        }
+
+        this.length = file.toFile().length();
+        this.scheduler = Schedulers.boundedElastic();
+    }
+
+    @Override
+    public Long getLength() {
+        return this.length;
+    }
+
+    @Override
+    public String toString() {
+        return new String(toBytes(), StandardCharsets.UTF_8);
+    }
+
+    @Override
+    public byte[] toBytes() {
+        bytes.compareAndSet(null, getBytes());
+        byte[] data = this.bytes.get();
+        return Arrays.copyOf(data, data.length);
+    }
```

Comment:
Thoughts on making these operations invalid on `FileContent`? Or is it fine to just have a OOM error happen if a file >2GB is used